fontforge/splineoverlap.c

```diff
@@ -3490,7 +3490,7 @@ static SplineSet *SSRemoveTiny(SplineSet *base) {
 
     while ( base!=NULL ) {
 	ssnext = base->next;
-	for ( sp=base->first; ; ) {
+	for ( sp=base->first; !( base == NULL || sp==base->first ); ) {
```

Comment:
With a for loop the guard is checked immediately after the initialization. At that point sp==base->first so the guard will be false. Therefore this addition just disables the loop.

Reply:
Thank you - I've closed this as now that you're looking into it I'm sure you'll come up with a better solution. :-)